bark_ml/library_wrappers/lib_tf_agents/runners/sac_runner.py

```diff
@@ -60,24 +81,35 @@ def GenerateExpertTrajectories(self, num_trajectories: int = 1000, render: bool
     per_scenario_expert_trajectories = {}
 
     while len(per_scenario_expert_trajectories) < num_trajectories:
-      expert_trajectories = {'obs': [], 'act': []}
+      expert_trajectories = {'obs_norm': [], 'obs': [], 'act': []}
 
+      # Merging Blueprint
+      # X: 881.707/1006.72
+      # Y: 1001.59/1010.82
+      # Theta: 0/6.28
+      # Velo: 0/50
       state = self._environment.reset()
-      expert_trajectories['obs'].append(state)
+      state_not_norm = self.GetStateNotNormalized()
+
+      expert_trajectories['obs_norm'].append(state)
+      expert_trajectories['obs'].append(state_not_norm)
       is_terminal = False
 
       while not is_terminal:
         action_step = self._agent._eval_policy.action(ts.transition(state, reward=0.0, discount=1.0))
 
         state, reward, is_terminal, info = self._environment.step(action_step.action.numpy())
-        expert_trajectories['obs'].append(state)
+        state_not_norm = self.GetStateNotNormalized()
+        expert_trajectories['obs_norm'].append(state)
+        expert_trajectories['obs'].append(state_not_norm)
         expert_trajectories['act'].append(action_step.action.numpy())
 
         if render:
           self._environment.render()
 
       if info and info['goal_reached']:
```

Comment:
I added the info check to filter the expert trajectories that end with goal reached. Only if the environment gives a valid last step (The workaround for the index error) and the goal was reached, we add the trajectory to the set of expert trajectories.
The change with `# if True` was just for debugging, so I didn't have to wait for a converged agent :)